### ISSUE TITLE
Bulk user invite processed with errors

### ISSUE THREAD

**Issue by ghost** [bitnami/bitnami-docker-discourse#33]
**Description**

I'm trying to add multiple users from csv file (previously exported from another discourse setup).
I run discourse using the bitnami [docker-compose.yml](https://github.com/bitnami/bitnami-docker-discourse/blob/master/docker-compose.yml) file.

**Steps to reproduce the issue:**

1. wget https://raw.githubusercontent.com/bitnami/bitnami-docker-discourse/master/docker-compose.yml
2. docker-compose up
3. Login to the discourse interface
4. Admin>Users>Send Invites>Bulk Invites from File

Then I receive a notification: `Bulk user invite processed with errors`
With the following error log: `[2017-11-10 14:19:32 +0000] Bulk Invite Process Failed -- 'No such file or directory @ rb_sysopen - /opt/bitnami/discourse-sidekiq/public/uploads/csv/default/user-list-171110-125927-3.csv'`.
The csv file does not exists under this path, I checked using docker exec.

Actually the csv has been uploaded in the container (the discourse main one, not the sidekiq) but under: `/bitnami/discourse/public/uploads/csv/default/user-list-171110-125927-3.csv`

Do I miss something? 

**Version**

- Output of `docker version`:

```
Client:
 Version:      17.05.0-ce
 API version:  1.29
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:08:20 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   89658be
 Built:        Thu May  4 22:08:20 2017
 OS/Arch:      linux/amd64
 Experimental: false

```

- Output of `docker info`:

```
Containers: 4
 Running: 4
 Paused: 0
 Stopped: 0
Images: 228
Server Version: 17.05.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 9048e5e50717ea4497b757314bad98ea3763c145
runc version: 9c2d8d184e5da67c95d601382adf14862e4f2228
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.10.14-100.fc24.x86_64
Operating System: Fedora 24 (Workstation Edition)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 7.656GiB
Name: douze-pc
ID: MSDR:3EOM:24KC:RSOT:7X3E:H65M:OOKF:XLAC:OQ7M:OOOY:23KV:6UTP
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.10.0, build 4bd6f1a
docker-py version: 2.0.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1t  3 May 2016
```

**Comment by dbarranco** [comment]
Hi @alex12Douze 

I have been able to reproduce this. 
This is happening due that Discourse-sidekiq and Discourse are using different volumes, so the information is not being shared between them and Discourse-Sidekiq is not able to read the uploads in the Discourse container.

In the meantime, you can fix this following this workaround:

* Log in the Discourse-Sidekiq contianer via terminal, in my case:

`docker exec -it 1_sidekiq_1 bash`

* Go to `/opt/bitnami/discourse-sidekiq/public/uploads/`:

`cd /opt/bitnami/discourse-sidekiq/public/uploads/`

* Create the following folders:

```
mkdir csv/
mkdir csv/default/
cd csv/default/
```

* Upload there the `.csv` file you pretend to upload to Discourse. You can do it using `docker cp` from out side the container.

* Give the correct owners to those files:

`sudo chown -R discourse:discourse /opt/bitnami/disckourse-sidekiq/public/uploads/`

> Note: Check that the folders and the `.csv` file you uploaded have `discourse:discourse` as owner:group.

* Then log in the admin panel of your Discourse deployment and go to Admin>Users>Send Invites>Bulk Invites from File.

* Upload the same `.csv` file.

Now both the Discourse container and Discourse-Sidekiq container will have the `.csv` file placed in the `uploads/` dir so it will work.

> Note: Do not upload the `.csv` file with two different names in both containers or it won't work.


I have created an internal task to solve this and to change the persistence of these containers to avoid this workaround to upload a csv file.
Thank you so much for reporting this!!

Best regards.

**Comment by dbarranco** [comment]
Hi @alex12Douze 

Another workaround for this kind of issue would be having the same volume for the Discourse and Discourse-Sidekiq container.
If you change the docker-compose to this one:

```
version: '2'
services:
  postgresql:
    image: 'bitnami/postgresql:9.6'
    volumes:
      - 'postgresql_data:/bitnami'
  redis:
    image: 'bitnami/redis:latest'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
    volumes:
      - 'redis_data:/bitnami'
  discourse:
    image: 'bitnami/discourse:latest'
    labels:
      kompose.service.type: nodeport
    ports:
      - '80:3000'
    depends_on:
      - postgresql
      - redis
    volumes:
      - 'discourse_data:/bitnami'
    environment:
      - POSTGRESQL_HOST=postgresql
      - POSTGRESQL_ROOT_USER=postgres
      - POSTGRESQL_CLIENT_CREATE_DATABASE_NAME=bitnami_application
      - POSTGRESQL_CLIENT_CREATE_DATABASE_USERNAME=bn_discourse
      - POSTGRESQL_CLIENT_CREATE_DATABASE_PASSWORD=bitnami1
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
  sidekiq:
    image: 'bitnami/discourse:latest'
    depends_on:
      - discourse
    volumes:
      - 'discourse_data:/bitnami'
    command: 'nami start --foreground discourse-sidekiq'
    environment:
      - DISCOURSE_POSTGRESQL_NAME=bitnami_application
      - DISCOURSE_POSTGRESQL_USERNAME=bn_discourse
      - DISCOURSE_POSTGRESQL_PASSWORD=bitnami1
      - DISCOURSE_HOST=discourse
      - DISCOURSE_PORT=3000
volumes:
  postgresql_data:
    driver: local
  redis_data:
    driver: local
  discourse_data:
    driver: local
```

In the Discourse and Discourse-Sidekiq you will have in `/bitnami` the persisted folders.
So you can upload the `.csv` to the Discourse container and do this:


* Create the appropriate folders:
```
mkdir /bitnami/discourse-sidekiq/public/uploads/csv/
mkdir /bitnami/discourse-sidekiq/public/uploads/csv/default/
chown -R discourse:discourse /bitnami/discourse-sidekiq/public/uploads/
```

* Go to the Discourse interface and upload the `.csv` file. After uploading that file you will have the `.csv` file in `/bitnami/discourse/public/uploads/csv/default/file.csv`

* Copy the files from the public folder from Discourse to the public folder of Discourse-Sidekiq

`sudo cp -rf /bitnami/discourse/public/uploads/csv/* /bitnami/discourse-sidekiq/public/uploads/csv/`

* Re-upload the file using  the Discourse interface and it will work.

I will leave up to you to choose the workaround you prefer. As I posted before, we will improve this to not to have to apply this kind of workarounds and make this easier the next time any user has to do this.

Please, do not hesitate to ask if you have any kind of question.

Kind regards!!

**Comment by beltran-rubo** [comment]
This issue has been solved in the version 2.0.3-r7. See this for more details:

https://github.com/bitnami/bitnami-docker-discourse/issues/40#issuecomment-410186599
